File: test/page-objects/streams/UI/eventHeader.ts
import Page from "../../page"
import reporter from "../../../utils/reporter"


class EventHeader extends Page {

    public static eventLink: any

    constructor() {
        super()
    }

    // Event header details
    get waitingIndicator() { return $(`//*[name()='svg' and @class='waiting-indicator']`) }
    get eventPublishButton() { return $(`//button[@class='small primary publish-btn' or @class='small primary']`) }
    get publisiModal() { return $(`//div[@class='modal normal']`) }
    get nextPublishButton() { return $(`(//button[@class='primary'])[1]`) }
    get assistantChecklistModal() { return $(`(//div[@class='modal normal'])[2]`) }
    get nextAssistantModalPublishButton() { return $(`(//button[@class='primary'])[2]`) }
    get copyEventLink() { return $(`//div[@class='copy-event reverse-tooltip']//button`) }


    async publishEvent() {
        try {
            if (await this.waitingIndicator.isDisplayed()) {
                await browser.pause(5000)
                reporter.addStep("info", 'Waiting to display Publish Event button from Loading state.')
            }
            await this.click(await this.eventPublishButton)
            reporter.addStep("info", `Clicked on the Publish Event button.`)
            await browser.pause(2000)
            if (await this.publisiModal.isDisplayed()) {
                await this.click(await this.nextPublishButton)
                reporter.addStep("info", 'Clicked on the next publish button')
            }
            await browser.pause(2000)
            if (await this.assistantChecklistModal.isDisplayed()) {
                await this.click(await this.nextAssistantModalPublishButton)
                reporter.addStep("info", 'Clicked on the next assitant modal publish button')
                await browser.pause(2000)
            }
        } catch (err) {
            err.message = `Error publishing the event: , ${err.message} `
            throw err
        }
    }

    async copyPublishedEventLink(): Promise<any> {
        try {
            if (await this.copyEventLink.isDisplayed()) {
                await this.click(await this.copyEventLink)
                reporter.addStep("info", 'Clicked on copy event link.')
            }
            var nodeClipboard = require('node-clipboardy')
            EventHeader.eventLink = await nodeClipboard.readSync()
            reporter.addStep("info", `The Event link is:  ${EventHeader.eventLink}`);
            return EventHeader.eventLink
        } catch (err) {
            err.message = `Error copying the vent link: , ${err.message} `
            throw err
        }
    }
}
export default new EventHeader

File: src/main/java/com/brandlive/qa/automation/events3/pages/admin/EventHeader.java
package com.brandlive.qa.automation.events3.pages.admin;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import com.brandlive.qa.automation.utils.Base;

import java.awt.*;
import java.awt.datatransfer.Clipboard;
import java.awt.datatransfer.DataFlavor;
import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;

public class EventHeader extends Base {

    public void publishEvent() {
        String waitingIndicator = "//*[name()='svg' and @class='waiting-indicator']";
        if (elementPresent(By.xpath(waitingIndicator))) {
            sleep(5);
            log.info("Waiting to display Publish Event button from Loading state.");
        }
        WebElement publishButton = webDriver.findElement(By.xpath("//button[@class='small primary publish-btn' or @class='small primary'] "));
        publishButton.click();
        WebElement publishModal = webDriver.findElement(By.xpath("//div[@class='modal normal']"));
        sleep(3);
        if (publishModal.isDisplayed()) {
            WebElement nextButton = webDriver.findElement(By.xpath("(//button[@class='primary'])[1]"));
            nextButton.click();
        }
        String assistantChecklistModal = "//div[@class='modal normal']";
        sleep(5);
        if (elementPresent(By.xpath(assistantChecklistModal))) {
            WebElement nextButton = webDriver.findElement(By.xpath("(//button[@class='primary'])[2]"));
            nextButton.click();
            log.info("The Assistant pop up is present and clicked on publish.");
            sleep(3);
        } else {
            log.info("The Assistant modal pop up is not present.");
        }
    }

    public String copyPublishedEventLink() {
        sleep(5);
        log.info("Copy the published event link...");
        WebElement copyEventLink = webDriver.findElement(By.xpath("//div[@class='copy-event reverse-tooltip']//button"));
        if (copyEventLink.isDisplayed()) {
            copyEventLink.click();
            log.info("Copied the published event link.");
        }

        log.info("Get the copied URL from the Clipboard...");
        Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
        String copiedEventURL = null;

        try {
            copiedEventURL = (String) clipboard.getContents(null).getTransferData(DataFlavor.stringFlavor);
        } catch (UnsupportedFlavorException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
        log.info("The copied event link is" + copiedEventURL);
        return copiedEventURL;
    }

    public void clickOnHeaderPathingArrow() {
        String pathingArrow = "//div[@class='pathing-arrow']";
        WebElement headerPathingArrow = waitForDisplayedAndEnabled(By.xpath(pathingArrow), "Header Pathing Arrow");
        if (headerPathingArrow.isDisplayed()) {
            headerPathingArrow.click();
        }
    }

    public void goToAdminPage() {
        String adminLink = "//img[@alt='Go to admin page']";
        WebElement adminLinkPage = waitForDisplayedAndEnabled(By.xpath(adminLink), "Clicked on the admin link page.");
        if (adminLinkPage.isDisplayed()) {
            adminLinkPage.click();
            log.info("Clicked on the Admin link page.");
        }
    }
}
